src/search/bucket/addresult.go

```diff
@@ -8,8 +8,27 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-// passing pointers down is on stack
-func AddSEResult(seResult *result.RetrievedResult, seName engines.Name, relay *Relay, options engines.Options, pagesCol *colly.Collector) {
+// Checks if the retrieved result is valid. Makes a result object and adds it to the relay.
+// Returns true if the result is valid (false otherwise).
+func AddSEResult(seResult *result.RetrievedResult, seName engines.Name, relay *Relay, options *engines.Options, pagesCol *colly.Collector) bool {
```

Comment:
options should stay pass-by-copy to ensure adding a result doesn't change it for other results, it's also a small struct consisting of 4 bools, 2 int and 3 strings.